CMakeLists.txt

```diff
@@ -1,5 +1,45 @@
-project(flash-attention LANGUAGES CXX CUDA)
-
-set(CUTLASS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cutlass)
-
-add_subdirectory(src)
+cmake_minimum_required(VERSION 3.12)
+
+project(flash-attention LANGUAGES CXX CUDA)
+
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --expt-relaxed-constexpr --use_fast_math -t 8 \
+                      -gencode=arch=compute_90a,code=\\\"sm_90a,compute_90a\\\" \

```

Comment:
90a compilation needs to be done conditionally. We still need to support A100 and A10g.